### PR TITLE
Fix rounding

### PR DESCRIPTION
**Pull request by atomiks**
As mentioned in #551, this will allow for "pixel-perfect" positioning if desired, since subpixel `top/left` positioning doesn't cause blurriness unlike `translate3d`.

### CODE REVIEW

**Comment by atomiks** [comment]
After doing some tests I think I have a decent solution to ensure precision when using `gpuAcceleration` too.

Allow people to specify a `round()` callback function which gets supplied the current `placement` and uses the return value. Browsers will imperfectly render the position relative to the reference sometimes, no matter what, so we need to manually decide which rounding to choose based on the placement (and maybe other additional checks like which browser/rendering engine is currently being used), and also the specific individual instance.

Something like this:

```js
new Popper(reference, popper, {
  modifiers: {
    computeStyle: {
      round: placement => Math.floor
    }
  }
})

// In source:
const floor = gpuAcceleration 
  ? find(
    data.instance.modifiers, 
    modifier => modifier.name === 'computeStyle'
  ).round(data.placement) 
  : v => v;
```

**Comment by atomiks** [comment]
It seems like top/bottom prefers `round` while left/right prefers `floor`. See:

round top/bottom, floor left/right:

rounding
rounding

all floor (sits too high vertically)

rounding

all round (sits too far horizontally)

rounding

If I change the width from 100x100 px to any combination of even/odd the floor + round combo seems to work every time. I also tried in FF and Safari with the same results.

**Comment by FezVrasta** [comment]
Oh, that's what I wanted to see! I knew there was a magic recipe! Thanks!

File: packages/popper/src/modifiers/computeStyle.js [hunk 1]
```diff
@@ -37,11 +37,11 @@ export default function computeStyle(data, options) {
     position: popper.position,
   };
 
-  // floor sides to avoid blurry text
+  // round sides to avoid blurry text
```
Comment (FezVrasta) [inline packages/popper/src/modifiers/computeStyle.js]: Care to improve this comment to explain the reasoning behind floor vs round?

**Comment by FezVrasta** [comment]
Great thanks, we just need tests to pass now

**Pull request merged**